cli/cmd/cluster/certificate-rotator.go

```diff
@@ -0,0 +1,244 @@
+package cluster
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kinvolk/lokomotive/pkg/k8sutil"
+	"github.com/kinvolk/lokomotive/pkg/platform"
+)
+
+type certificateRotatorConfig struct {
+	clientSet            *kubernetes.Clientset
+	newCACert            string
+	logger               *log.Entry
+	daemonSetsToRestart  []platform.Workload
+	deploymentsToRestart []platform.Workload
+}
+
+type certificateRotator struct {
+	config certificateRotatorConfig
+}
+
+const (
+	// Time to wait between updating DaemonSet/Deployment and start looking if
+	// workload has converged. This should account for kube-controller-manager election time (default: 15s),
+	// reconciliation period and time spent in reconciliation loop (based on number of workloads
+	// in the cluster).
+	kubeControllerManagerReconciliationPeriod = 20 * time.Second
+)
+
+type CertificateRotateOptions struct {
+	Confirm    bool
+	Verbose    bool
+	ConfigPath string
+	ValuesPath string
+}
+
+func RotateCertificates(contextLogger *log.Entry, options CertificateRotateOptions) error {
+	cc := clusterConfig{
+		verbose:    options.Verbose,
+		configPath: options.ConfigPath,
+		valuesPath: options.ValuesPath,
+	}
+
+	c, err := cc.initialize(contextLogger)
+	if err != nil {
+		return fmt.Errorf("initializing: %w", err)
+	}
+
+	exists, err := clusterExists(c.terraformExecutor)
+	if err != nil {
+		return fmt.Errorf("checking if cluster exists: %w", err)
+	}
+	if !exists {
+		return errors.New("cannot rotate certificates on a non-existent cluster")
+	}
+	if c.platform.Meta().Managed {
+		// TODO: do we want to error here?
+		return errors.New("cannot rotate certificates on a managed cluster")
```

Comment:
We can move all this into separate function named `rotatableCluster` or something :)